mod_ci/controllers.py

```diff
@@ -596,6 +596,14 @@ def progress_reporter(test_id, token):
             if request.form['type'] == 'progress':
                 # Progress, log
                 status = TestStatus.from_string(request.form['status'])
+                # Check whether test is not running previous status again
+                istatus = TestStatus.progress_step(status)
+                if len(test.progress) != 0:
+                    laststatus = TestStatus.progress_step(
+                        test.progress[-1].status)
+                    if laststatus > istatus:
+                        status = TestStatus.canceled
```

Comment:
This might add a canceled status to the platform, but this won't stop the CI from running.